src/test/java/seedu/dengue/testutil/PersonListBuilder.java

```diff
@@ -0,0 +1,152 @@
+    }
+
+    /**
+     * Returns a new {@code PersonListBuilder} which builds a {@code List} of {@code Person} instances with
+     * postal codes following the list provided, but are otherwise identical to the default {@code Person}.
+     *
+     * @param postals The desired postals of the people in the output list.
+     * @return A new list builder.
+     */
+    public PersonListBuilder withPostals(List<String> postals) {
+        List<Person> personList = new ArrayList<>();
+        Person defaultPerson = this.personBuilder.build();
+
+        postals.forEach(postal -> personList.add(this.personBuilder.withPostal(postal).build()));
+
+        return new PersonListBuilder(defaultPerson, personList);
+    }
+
+    /**
+     * Returns a new {@code PersonListBuilder} which builds a {@code List} of {@code Person} instances with
+     * case dates following the list provided, but are otherwise identical to the default {@code Person}.
+     *
+     * @param dates The desired dates of the cases in the output list.
+     * @return A new list builder.
+     */
+    public PersonListBuilder withDates(List<String> dates) {
+        List<Person> personList = new ArrayList<>();
+        Person defaultPerson = this.personBuilder.build();
+
+        dates.forEach(date -> personList.add(this.personBuilder.withDate(date).build()));
+
+        return new PersonListBuilder(defaultPerson, personList);
+    }
+
+    /**
+     * Returns a new {@code PersonListBuilder} which builds a {@code List} of {@code Person} instances with
+     * dengue variants following the list provided, but are otherwise identical to the default {@code Person}.
+     *
+     * @param variants The desired list of dengue variants (possibly empty) contracted by the people in the output list.
+     * @return A new list builder.
+     */
+    public PersonListBuilder withVariants(List<String[]> variants) {
+        List<Person> personList = new ArrayList<>();
+        Person defaultPerson = this.personBuilder.build();
+
+        variants.forEach(variantArr -> personList.add(this.personBuilder.withVariants(variantArr).build()));
+
+        return new PersonListBuilder(defaultPerson, personList);
+    }
+
+    /**
+     * Returns a {@code List} of {@code Person} instances, according to the list of people stored.
+     *
+     * @return The desired list of people.
+     */
+    public List<Person> build() {
+        return new ArrayList<>(this.personList);
+    }
+
+    /**
+     * Returns a {@code List} containing only one default {@code Person} instance.
+     *
+     * @return A singleton containing the default person.
+     */
+    public List<Person> buildSingle() {
+        return List.of(this.personBuilder.build());
+    }
+
+    /**
+     * Returns a {@code List} containing no {@code Person} instances.
+     *
+     * @return An empty list.
+     */
+    public List<Person> buildEmpty() {
+        return new ArrayList<>(EMPTY_PERSON_LIST);
+    }
+}
```

Comment:
`private static final List<Person> EMPTY_PERSON_LIST = new ArrayList<>();`
` return new ArrayList<>(EMPTY_PERSON_LIST);`
Thats a wrapper around a wrapper
```suggestion
    /**
     * Returns a {@code List} containing no {@code Person} instances.
     *
     * @return An empty list.
     */
    public List<Person> buildEmpty() {
        return new ArrayList<>();
    }
}
```

Reply:
LOL I've resolved this. Thanks.